### ISSUE TITLE
Not backward-compatible with actual spotipy

### ISSUE THREAD

**Issue by stephanebruckert** [felix-hilden/spotipy#98]
I saw your repo is being discussed in https://github.com/plamere/spotipy/issues/387 as a spotipy alternative. 

I started importing it in my project in order to use `playlist_cover_image_upload` but then I realised that your version has nothing in common with the original spotipy.

If this repo is backward-incompatible with spotipy, I don't think it should be named spotipy as it can be misleading (mainly because the name is already used on PyPI, and the doc is already a top result on google https://spotipy.readthedocs.io/

**Comment by stephanebruckert** [comment]
Just saw that this has been mentioned here already https://github.com/plamere/spotipy/issues/387#issuecomment-520859488

+1 for spotipy3!

**Comment by felix-hilden** [comment]
Thank you for the feedback, have a look at #22 for the rationale behind this change and voice your opinions about them, please! I'd love to hear your thoughts as well. We have plans of updating the package and documentation gracefully. But I'll close this issue as a duplicate.

To paraphrase, in addition to this really being based on the original spotipy, the old one is also quite broken. So I don't see any value in keeping it. Creating a new version would solve issues for lots of users. And, I think the original repository has no `playlist_cover_image_upload`... I'm not sure what you mean.

**Comment by stephanebruckert** [comment]
1. My app uses Spotipy to do a bunch of things, everything works!
2. I want to add a quick feature to upload thumbnails but spotipy does not have this endpoint
3. I upgrade spotipy to this/your "new version" that seems to have the thumbnails endpoint
4. I notice that my app breaks on initialising spotipy, it's unclear why at first, everything was working fine before, what happened? I was hoping that this new version would benefit me, but it is in fact requiring me to modify most parts of my app. I give up because it adds unnecessary risks:
   - losing development time,
   - adding bugs to my own code,
   - expect possible regressions in the new library itself. Should I trust it? it doesn't seem very used nor have a lot of contributors, how do I know it will do the job?
5. I find that a more efficient alternative for me is to rather fork the original spotipy and implement the thumbnail endpoint [on my own](https://github.com/plamere/spotipy/compare/master...mirrorfm:master).

Basically as a user, I am looking for a solution to a specific problem, not for a solution to a problem I don't have.

I hope this clarifies the importance of backward-compatibility?

**Comment by felix-hilden** [comment]
That's not really a surprise, is it :D You must have gotten here from some of the initial discussions, read them, seen the readme and documentation, and concluded that it is not just a fork with a couple of typos fixed. After that you surely have considered is modifying your code worth it for you or should you use some other fork like Harrison's, not blindly just dropped my version in. I'm sorry if this all seems aggressive, but I can't imagine you just seeing a link to this repository, cloning it and installing it, and never being informed that it's not exactly a drop-in replacement. If that was the case, did you look for information? If you did, where did you not find it? Maybe the information could be there in the future.

You could trust my version, based on running your own tests, my tests, examining the code coverage and reading example scripts. But yes, a more efficient alternative for you **is** writing the endpoint on your own, if you are simply interested in having one endpoint more and keeping to the old version. I'm interested in providing an improved solution.
